### ISSUE TITLE
Irrelevant named tensor warnings

### ISSUE THREAD

**Issue by h6197627** [pytorch/pytorch#54846]
## 🐛 Bug
This code:
```
import torch
import torch.nn as nn

input_tensor = torch.ones((1, 32, 128, 128), dtype=torch.float)
cs = nn.ChannelShuffle(groups=2)
cs(input_tensor)
```

produces irrelevant warning:

> /home/user/PyTorch/dist-packages/release/torch/nn/modules/channelshuffle.py:46: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ../c10/core/TensorImpl.h:973.)
>   return F.channel_shuffle(input, self.groups)

as no named tensors are used.

## Expected behavior

No warnings

## Environment

 - PyTorch Version (e.g., 1.0): current master
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): source

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @mruberry

**Comment by jbschlosser** [comment]
Hey @h6197627, thanks for reporting this! `nn.ChannelShuffle` uses named tensor logic internally, resulting in the warning:
https://github.com/pytorch/pytorch/blob/7d56de18345bfaedf31cbb400389cbd6111b216d/aten/src/ATen/native/ChanelShuffle.cpp#L55

I agree this warning is confusing and will look into options for a fix.

**Comment by zou3519** [comment]
Fix should be just to replace
```
return namedinference::propagate_names_if_nonempty(
    output_tensor,
    self.names());
```
with
```
return namedinference::propagate_names_if_nonempty(
    output_tensor,
    self.has_names() ? self.names() : {});
```

**Comment by ezyang** [comment]
don't do a std::vector; ArrayRef is good enough

**Comment by BramVanroy** [comment]
I encounter this warning as well on `torch==1.9.0` (Windows, Python 3.8, cpu) when using maxpool1d. Specifically, this line:

https://github.com/stanfordnlp/stanza/blob/166e80e00254b1bf97961dc3462d043e0a0590c3/stanza/models/classifiers/cnn_classifier.py#L364

triggers this warning:

> site-packages\torch\nn\functional.py:652: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  ..\c10/core/TensorImpl.h:1156.)
>   return torch.max_pool1d(input, kernel_size, stride, padding, dilation, ceil_mode)

**Comment by glenn-jocher** [comment]
This warning also appears in YOLOv5 now with torch 1.9.0. If this warning serves no purpose then I think it should be removed so as not to unduly worry users.

To reproduce run Setup and Inference cells in YOLOv5 notebook: https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb

```
/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
```

<img width="1108" alt="Screenshot 2021-06-25 at 13 53 12" src="https://user-images.githubusercontent.com/26833433/123421151-c6c06e80-d5bc-11eb-96ec-b9f03f6cb138.png">

**Comment by zou3519** [comment]
We've fixed the problems with max_pool{1, 2, 3}d on master with https://github.com/pytorch/pytorch/pull/60059. The fix will be in our next release.

**Comment by glenn-jocher** [comment]
@zou3519 great, thank you!
